pydra/engine/tests/test_shelltask_inputspec.py

```diff
@@ -2073,3 +2073,412 @@ def test_shell_cmd_inputs_di(tmpdir, use_validator):
+                attr.ib(
+                    type=File,
+                    metadata={
+                        "help_string": "fictional output #1",
+                        "output_file_template": "{out_dir}/test_1.nii",
+                        "absolute_path": True,
+                    },
+                ),
+            ),
+            (
+                "out_2",
+                attr.ib(
+                    type=File,
+                    metadata={
+                        "help_string": "fictional output #1",
+                        "output_file_template": "{out_dir}/test.nii",
+                        "absolute_path": True,
+                    },
+                ),
+            ),
+        ],
+        bases=(ShellOutSpec,),
+    )
+
+    shelly = ShellCommandTask(
+        executable="touch", input_spec=input_spec, output_spec=out_spec, out_dir=tmpdir
+    )
+    shelly()
+    res = shelly.result()
+    # checking if the output was created
+    assert (Path(tmpdir) / Path("test.nii")).exists()
+    # check if path of the out_1 is correct
+    assert res.output.out_1 == attr.NOTHING
+    # check if path of the out_1 is correct
+    assert res.output.out_2 == (Path(tmpdir) / Path("test.nii"))
+
+
+def test_shellspec_cmd_absolute_path_non_existing_multi(tmpdir):
+    """test the 'absolute_path' metadata option. An output path marked with this should not be appended
+    to the nodes output_dir. Testing for a existing and non existing file in multiOutputObj."""
+    input_spec = SpecInfo(
+        name="Input",
+        fields=[
+            (
+                "out_name",
+                attr.ib(
+                    type=MultiInputObj,
+                    metadata={
+                        "help_string": """
+                        base name of the pretend outputs.
+                        """,
+                        "mandatory": True,
+                        "argstr": "...",
+                    },
+                ),
+            )
+        ],
+        bases=(ShellSpec,),
+    )
+    out_spec = SpecInfo(
+        name="Output",
+        fields=[
+            (
+                "out_list",
+                attr.ib(
+                    type=MultiOutputFile,
+                    metadata={
+                        "help_string": "fictional output #1",
+                        "output_file_template": "{out_name}",
+                        "absolute_path": True,
+                    },
+                ),
+            ),
+        ],
+        bases=(ShellOutSpec,),
+    )
+
+    shelly = ShellCommandTask(
+        executable="touch",
+        input_spec=input_spec,
+        output_spec=out_spec,
+        out_name=[str(Path(tmpdir) / "test_1.nii"), "test_2.nii"],
+    )
+    shelly()
+    res = shelly.result()
+    # checking if the outputs are Nothing
+    assert res.output.out_list[0] == Path(tmpdir) / "test_1.nii"
+    assert res.output.out_list[1] == attr.NOTHING
+
+
+def test_shellspec_formatter_1(tmpdir):
```

Comment:
we could use a `pytest.mark.parametrize` to split this test

Reply:
btw. some of the test could be also represented by `readonly` option, as in `test_shell_cmd_inputs_di`, but we discussed about it. Your approach is more general

Reply:
Yes, another use case to test could be the following. An application created either an output with extension '.nii.gz' or '.nii' depending on a flag. So, the ```formatter``` could be used to create the right template based on the flag.